Add Kyverno Project Graduation Due Diligence (#2057)
cncf/toc: projects/kyverno/kyverno-graduation-due-diligence.md

* Addressed the previous maintainer ladder concerns and continue to evaluate organizational and maintainer diversity
* Granular policy enforcement to help manage specific security deviations in complex environments
* Extensive documentation allowing easy and fast implementation
* Operational stability at scale with reliability required by large adopting organizations
* Useful subprojects such as [Kyverno Chainsaw](https://github.com/kyverno/chainsaw) that enables declarative end-to-end testing in a Kubernetes cluster

The following actions were provided to the project that were considered blocking but since resolved:
* Build a robust maintainer ladder and update governance - and show in action for at least 6 months - to address organizational diversity concerns. See Governance Section.
* [Update CODEOWNERS file to reflect accurate maintainer list](https://github.com/kyverno/kyverno/issues/15110)
* [Vendor Neutrality: Remove references to the contributing organization](https://github.com/kyverno/kyverno/issues/15113)
* [Link Release Procedure to main Release page](https://github.com/kyverno/website/issues/1867)

> * [Update the ROADMAP file to include the Roadmap change process](https://github.com/kyverno/website/issues/1907)

Reviewer: ```suggestion
* [Update the ROADMAP file to include the Roadmap change process](https://github.com/kyverno/kyverno/issues/15478)
```

  The [community page](https://kyverno.io/community/) lists meeting times and links to agendas stored on Google Docs.

  Meetings are also published on [the CNCF calendar](https://calendar.cncf.io).
  

- [x] **Documentation of how to contribute, with increasing detail as the project matures.**

  Kyverno's general contributing docs are outlined in [CONTRIBUTING.md](https://github.com/kyverno/community/blob/cfa604580b15f3f971d6c7b5b4bc395a758940e2/CONTRIBUTING.md) within the [kyverno/community](https://github.com/kyverno/community) repo.
  
  This is augmented by subproject specific contributor guides in each other repo.
   

- [x] **Demonstrate contributor activity and recruitment.**

  New contributors are listed with each [release](https://github.com/kyverno/kyverno/releases). For example the [v1.17 release](https://github.com/kyverno/kyverno/releases/tag/v1.17.0) had 14 new contributors.

  Additional maintainer recruitment can be seen in the commit history of [MAINTAINERS.md](https://github.com/kyverno/community/commits/cfa604580b15f3f971d6c7b5b4bc395a758940e2/MAINTAINERS.md).

## Engineering Principles

> Note: This report is augmented by a [General Technical Review](https://github.com/cncf/toc/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E) that was performed by Dylan Page in early 2026.

Reviewer: ```suggestion
Note: This report is augmented by a [General Technical Review](https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E) that was performed by Dylan Page in early 2026.
```

### Suggested

N/A

### Required

- [x] **Document project goals and objectives that illustrate the project’s differentiation in the Cloud Native landscape as well as outlines how this project fulfills an outstanding need and/or solves a problem differently. _This requirement may also be satisfied by completing a General Technical Review._**

>   * A General Technical Review was completed on 02-FEB-2026, and can be discovered at [https://github.com/cncf/toc/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E](https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E).

Reviewer: ```suggestion
  * A General Technical Review was completed on 02-FEB-2026, and can be discovered at [https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E](https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E).
```

 

- [x] **Document what the project does, and why it does it - including viable cloud native use cases. _This requirement may also be satisfied by completing a General Technical Review._**

>   * A General Technical Review was completed on 02-FEB-2026, and can be discovered at [https://github.com/cncf/toc/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E](https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E).

Reviewer: ```suggestion
  * A General Technical Review was completed on 02-FEB-2026, and can be discovered at [https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E](https://github.com/cncf/toc/blob/main/projects/kyverno/tech-review/2026-02-02.md%E2%80%8E).
```

- **Responsive Community:** The maintainers were commended for their quick response times in fixing critical bugs, particularly those related to admission review latency and memory utilization.

**Areas for Improvement**

- **Regression & Performance Testing:** 
  - **Internal Regressions:** There is a need for more robust internal testing to catch regressions before release (e.g., the instance where a metrics refresh flag stopped functioning after a migration).
  - **Proactive Scale Testing:** Rather than relying on large-scale users to find limits, the project should perform more proactive, high-scale performance testing internally.
- **Documentation for Advanced Use Cases:**
  - **Complex Logic:** There is a lack of documentation for complex, real-world scenarios, such as enforcing pod-level policies based specifically on namespace labels.
  - **Block Nuance:** The documentation needs to better clarify the functional differences and best practices for using validate vs. deny blocks.

> Administrative User Experience: While policy authoring is "low-code," the administrative experience of managing Kyverno at scale is still perceived as requiring "power user" expertise. Simplifying the operational management of the controllers would be beneficial.

Reviewer: ```suggestion
- Administrative User Experience: While policy authoring is "low-code," the administrative experience of managing Kyverno at scale is still perceived as requiring "power user" expertise. Simplifying the operational management of the controllers would be beneficial.
```
